packages/react-button/src/components/Button/Button.types.tsx

```diff
@@ -1,5 +1,14 @@
 import * as React from 'react';
 import { ComponentProps, ShorthandValue, ComposeOptions, ComposeStandardStatics } from '../../utils/tempTypes';
+import { TokenSetType } from '@fluentui/react-theme-provider';
+
+export type RecursivePartial<T> = {
+  [P in keyof T]?: T[P] extends (infer U)[]
+    ? RecursivePartial<U>[]
+    : T[P] extends object
+    ? RecursivePartial<T[P]>
+    : T[P];
+};
```

Comment:
This is just a temporary location right? I feel like this should be in the `compose` typings.

Reply:
will need to move it. I have i think 3 copies now.

Reply:
moved to temptypes.